sklearn/tests/test_common.py

```diff
@@ -231,3 +232,13 @@ def test_get_params_invariance():
             else:
                 yield _named_check(
                     check_get_params_invariance, name), name, Estimator
+
+
+@ignore_warnings
+def test_predict_does_not_change_estimator_state():
```

Comment:
I think you should add this test to the generator that generates all tests in ``estimator_checks.py`` and not add a test here. Though for working on it this might be easier.